src/main/java/net/glowstone/io/entity/EntityStore.java

```diff
@@ -98,6 +102,37 @@ public void load(T entity, CompoundTag tag) {
             UUID uuid = UUID.fromString(tag.getString("UUID"));
             entity.setUniqueId(uuid);
         }
+
+        if (tag.isList("Passengers", TagType.COMPOUND)) {
+            for (CompoundTag entityTag : tag.getCompoundList("Passengers")) {
+                Entity passenger = loadPassenger(entity, entityTag);
+                if (passenger != null) {
+                    entity.addPassenger(passenger);
```

Comment:
Perhaps you should make another ("internal") method when setting the passengers from NBT -- you wouldn't want to fire these events when loading the world.